### PR TITLE
Feature/shreyagupta112/user routes

### PR DESCRIPTION
**Pull request by shreyagupta112**
### Administrative Info
#38 
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
I created the auth_header.js file and modified the UserRoutes.js file to include new user routes.  I created the auth_header.js file to used by a few user routes.  I created routes to upgrade users to admin and verified by id, a route to get a user's profile by id, and a route to update user information.  The route to upgrade a user to admin requires the logged-in user to be an admin and the route to update user information requires either the logged-in user to be an admin or to be the same user as the one who's information is being edited.

### Testing
How did you confirm your changes worked? 
I used Postman to run each of the user routes.  

### Confirmation of Change 
I created the auth_header.js file.  Everything that isn't the  secure and admin routes has been changed.

| [TODO Screenshot Title] |
| --- |
| [TODO Screenshot Link] |

### CODE REVIEW

**Review by PatrickBrown1 — APPROVED** [review]

File: backend/routes/UserRoutes.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
+const mongoose = require("mongoose");
 
 const router = express.Router();
-const { UserModel } = require("../models/UserModel");
 
+const UserModel = require("../models/UserModel");
+const { idOfCurrentUser } = require("../util/userUtil");
 // temporary secure route, accessed with /users/
+
 router.get("/secure", (req, res, next) =>
```
Comment (mohakvni) [inline backend/routes/UserRoutes.js:10]: Since no longer needed, can be removed!

**Review by mohakvni — DISMISSED** [review]
Just to sanitize a bit, we can remove the "/secure" route in UserRoutes otherwise everything else, LGTM

**Review by mohakvni — DISMISSED** [review]
LGTM

**Comment by mohakvni** [comment]
So as of now I have found the following errors:
    1. When we try to find user details of user whose Id does not exist in the database, server crashes
    2. When we try to verify a non existent user (ID not in database), does not display appropriate error               but also server does not crash
    3. Server crashed when trying to promote admin of User whose ID not in database
    4. Edit info automatically sets everything to null, if none of the information is passed in request body
    5. Sever crashed when we try to edit information fo non existent ID

**Review by mohakvni — APPROVED** [review]
Tested All possible edge cases, work right about fine! 
LGTM

**Pull request merged**